cpp/include/cudf/utilities/legacy/wrapper_types.hpp

```diff
@@ -77,15 +77,15 @@ struct wrapper
   using value_type = T;                                          ///< The underlying fundamental type of the wrapper
   value_type value;                                              ///< The wrapped value
 
-  template <gdf_dtype dtype = type_id,
+  template <typename from_type, gdf_dtype dtype = type_id,
```

Comment:
This will make _all_ wrappers constructible from any type, not just `value_type`, which is not what we want.